frontend-react/src/components/ManagePublicKey/ManagePublicKeyChooseSender.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useState } from "react";
+import { Form, Dropdown, Button, FormGroup } from "@trussworks/react-uswds";
+
+import { useOrganizationSenders } from "../../hooks/UseOrganizationSenders";
+import Spinner from "../Spinner";
+
+export interface ManagePublicKeyChooseSenderProps {
+    onSenderSelect: (sender: string) => void;
+}
+
+export default function ManagePublicKeyChooseSender({
+    onSenderSelect,
+}: ManagePublicKeyChooseSenderProps) {
+    const { isLoading, senders } = useOrganizationSenders();
+    const [selectedSender, setSelectedSender] = useState(
+        senders?.length === 1 ? senders[0].name : ""
+    );
+
+    useEffect(() => {
+        if (senders?.length === 1) {
+            onSenderSelect(selectedSender);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [senders]);
+
+    function handleSubmit() {
+        onSenderSelect(selectedSender);
+    }
+
+    return (
+        <div data-testid="ManagePublicKeyChooseSender">
+            {isLoading && <Spinner message="Loading..." />}
+            {!isLoading && senders && senders?.length > 1 && (
```

Comment:
Very minor point, but `senders && senders?.length > 1` is redundant; we can just simplify to `senders?.length > 1` since that'll always be false if `senders` is falsy.

Reply:
Needed to add since 'senders.length' is possibly 'undefined'.

Reply:
But wouldn't `senders?.length` (with the safe navigation/optional chaining via `?.`) already handle that?  Like:

```ts
const senders = undefined; // just for demonstration purposes
senders?.length > 1; // evaluates to false
```

Reply:
ok I will change the return value in `UseOrganizationSenders.ts` to return empty [] if no senders and that way I can apply `senders?.length > 1 `here.